lib/directive.js

```diff
@@ -134,6 +141,12 @@ function init(datetime, $log, $document){
 			}
 		}
 
+		if (angular.isDefined(attrs.datetimeTimezone)) {
```

Comment:
To make the directive dynamically responsive to the `timezone` variable, we have to do the following stuff inside `linkFunc`:

1. Assume the attribute is an expression and `$watch` the expression, or
2. Assume the attribute is dynamic (interpolated through `{{}}`) and `attrs.$observe` the attribute.

The first way is cleaner that users don't have to write additional `{{}}` for dynamic value, which is also how `datetime-utc` is implemented. The second way is used by `min`, `max` attributes that when the attribute is changed, it re-validates the input.

I suggest follow how `datetimeUtc` is implemented:
https://github.com/eight04/angular-datetime/blob/269ece740d284594ac7ab0a4baed4d837ec13780/lib/directive.js#L129-L135
```js
// check whether datetime-timezone is used
if (angular.isDefined(attrs.datetimeTimezone)) {
	if (/^[+-]\d{2}:?\d{2}$/.test(attrs.datetimeTimezone)) {
		// if the value looks like a timezone string e.g. datetime-timezone="+0700", datetime-timezone="+08:00"
		setTimezone(attrs.datetimeTimezone);
	} else {
		// if the value doesn't look like a timezone e.g. datetime-timezone="timezone"
		// assume it is an expression (a variable on $scope) and $watch it
		scope.$watch(attrs.datetimeUtc, setTimezone);
	}
}
```

Reply:
@eight04 awesome. Thank you for detailed answer!

Reply:
PR updated, all tests passed